pkg/controller/fed_backup_schedule_status_updater.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pingcap/tidb-operator/pkg/apis/federation/pingcap/v1alpha1"
+	informers "github.com/pingcap/tidb-operator/pkg/client/federation/informers/externalversions/pingcap/v1alpha1"
+	listers "github.com/pingcap/tidb-operator/pkg/client/federation/listers/pingcap/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog/v2"
+)
+
+// VolumeBackupScheduleStatusUpdaterInterface is an interface used to update the VolumeBackupScheduleStatus associated with a VolumeBackupSchedule.
+// For any use other than testing, clients should create an instance using NewRealBackupScheduleStatusUpdater.
+type VolumeBackupScheduleStatusUpdaterInterface interface {
+	// UpdateBackupScheduleStatus sets the backupSchedule's Status to status. Implementations are required to retry on conflicts,
+	// but fail on other errors. If the returned error is nil backup's Status has been successfully set to status.
+	UpdateBackupScheduleStatus(*v1alpha1.VolumeBackupSchedule, *v1alpha1.VolumeBackupScheduleStatus, *v1alpha1.VolumeBackupScheduleStatus) error
+}
+
+// NewRealVolumeBackupScheduleStatusUpdater returns a VolumeBackupScheduleStatusUpdaterInterface that updates the Status of a VolumeBackupScheduleStatus,
+// using the supplied client and bsLister.
+func NewRealVolumeBackupScheduleStatusUpdater(deps *BrFedDependencies) VolumeBackupScheduleStatusUpdaterInterface {
+	return &realVolumeBackupScheduleStatusUpdater{
+		deps: deps,
+	}
+}
+
+type realVolumeBackupScheduleStatusUpdater struct {
+	deps *BrFedDependencies
+}
+
+func (u *realVolumeBackupScheduleStatusUpdater) UpdateBackupScheduleStatus(
+	bs *v1alpha1.VolumeBackupSchedule,
+	newStatus *v1alpha1.VolumeBackupScheduleStatus,
+	oldStatus *v1alpha1.VolumeBackupScheduleStatus) error {
+
+	ns := bs.GetNamespace()
+	bsName := bs.GetName()
+	// don't wait due to limited number of clients, but backoff after the default number of steps
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		_, updateErr := u.deps.Clientset.FederationV1alpha1().VolumeBackupSchedules(ns).Update(context.TODO(), bs, metav1.UpdateOptions{})
+		if updateErr == nil {
+			klog.Infof("BackupSchedule: [%s/%s] updated successfully", ns, bsName)
```

Comment:
How about logging in with the `Volume` prefix (`VolumeBackupSchedule` ...)

Reply:
Fixed.